Add examples for SHOW CREATE FUNCTION statement (#22123)
trinodb/trino: docs/src/main/sphinx/routines/examples.md

SELECT answer() + 5; -- 47
```

> You can show SQL routines and plugin functions in a schema. This example uses

Reviewer: ```suggestion
You can show SQL catalog routines stored in a specific catalog and schema. The following example lists all routines in the `default` schema of the `example` catalog:
```

Reviewer: Did you forget to push or commit this change?

the `default` schema of the `example` catalog:

```sql
SHOW FUNCTIONS FROM example.default
```

> Of the functions returned, find all that match a character pattern. This example

Reviewer: Test if you can combine from and like and make an example with that.

Reviewer: Still need to test that

returns results with a name beginning with `ans`:

```sql
SHOW FUNCTIONS like 'ans%';
```

> Show the SQL statement that created the function. In this example the function is `answer`:

Reviewer: Add note that the () is not used and that if there are multiple routines with such a name but different parameters they are all returned.

SELECT answer() + 5; -- 47
```

You can show SQL catalog routines stored in a specific catalog and schema. The
following example lists all routines in the `default` schema of the `example`
catalog:

```sql

> SHOW FUNCTIONS FROM example.default

Reviewer: ```suggestion
SHOW FUNCTIONS FROM example.default;
```

```

Of the functions returned, find all that match a character pattern. This example
returns results with a name beginning with `ans`:

```sql

> SHOW FUNCTIONS like 'ans%';

Reviewer: ```suggestion
SHOW FUNCTIONS LIKE 'ans%';
```

```

> Show the SQL statement that created the function. In this example the function

Reviewer: Only works for catalog routines .. not functions. Test what happens for normal functions.

> is `answer`. To ensure all routines with the same name and different parameters

Reviewer: Its not about ensuring .. it just works that way .. you specify the name of the catalog routine only .. and then it returns one or more definitions.

are returned, parenthesis `()` are excluded.

```sql

> SHOW CREATE FUNCTION answer;

Reviewer: Show the output in this example
